### PR TITLE
6 - iio:libraries: Replace iio_interfaces with list

### PR DESCRIPTION
**Pull request by mchindri**
Only last commit is relevant
Built on top of: #666

### CODE REVIEW

File: iio/iio_demo/iio_demo.c [hunk 1]
```diff
@@ -452,7 +452,7 @@ int32_t iio_demo_init(struct iio_demo_desc **desc,
 		.write_data = iio_demo_write_dev,
```
Comment (cristiantpop) [inline iio/iio_demo/iio_demo.c:452]: Check commit message, it doesn't sound right.

File: libraries/iio/iio.c [hunk 2]
```diff
@@ -788,34 +760,17 @@ ssize_t iio_register(struct iio_desc *desc, struct iio_interface *iio_interface)
...
-			deleted = 1;
-			continue;
-		}
-		interfaces->interfaces[i - deleted] = iio_interfaces->interfaces[i];
-	}
-
-	interfaces->num_interfaces = iio_interfaces->num_interfaces - 1;
-	free(iio_interfaces);
+	/* Get if the item is found, get will remove it from the list */
+	return list_get_find(desc->interfaces_list, (void **)&to_remove_interface,
```
Comment (cristiantpop) [inline libraries/iio/iio.c:766]: Probably you need to free some memory for `to_remove_interface`

Comment (mchindri) [reply libraries/iio/iio.c:766]: to_remove_interface is allocated in the main application. Should be free there.
Unregister should just remove the entry from the list. This is what get function do.

File: iio/iio_demo/iio_demo.c (hunk 1 above)
Comment (mchindri) [reply iio/iio_demo/iio_demo.c:452]: done

**Comment by mchindri** [comment]
v2:
 - Fix commit message

**Comment by mchindri** [comment]
v3:
 Rebase on top of: #666 . Review only last commit

**Comment by mchindri** [comment]
v4: 
 rebase again

**Comment by mchindri** [comment]
v5:
 - rebase onto master
 - apply astyle

File: libraries/iio/iio.c [hunk 3]
```diff
@@ -47,25 +47,14 @@
 #include "ctype.h"
 #include "tinyiiod.h"
 #include "util.h"
+#include "list.h"
 #include "error.h"
-#include "errno.h"
```
Comment (mirceacaprioru) [inline libraries/iio/iio.c:51]: we should start using errno. Do not remove it.

Comment (mchindri) [reply libraries/iio/iio.c:51]: it was added to error.h

**Pull request merged**